packages/date-picker/src/vaadin-date-picker.js

```diff
@@ -222,11 +222,6 @@ class DatePicker extends DatePickerMixin(InputControlMixin(ThemableMixin(Element
 
   /** @private */
   _onVaadinOverlayClose(e) {
-    if (this._openedWithFocusRing && this.hasAttribute('focused')) {
-      this.setAttribute('focus-ring', '');
-    } else if (!this.hasAttribute('focused')) {
-      this.blur();
```

Comment:
This logic is removed without replacement because it wasn't working anyways (focus is restored after a timeout). Furthermore, the case of calling `input.blur()` is covered by `focus` listener:

https://github.com/vaadin/web-components/blob/9e022ac3f50fa9a40ab33c0cf0a5fa2a4bc5103b/packages/date-picker/src/vaadin-date-picker-mixin.js#L366-L371